From 897a5298b10c32f2ec15b8c2660e272e458c2cf6 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 25 Aug 2015 16:20:58 +0800
Subject: [PATCH] MLK-11411 ASoC: wm8994-core: add suspend/resume support

When the codec power is off in suspend funtion, the codec register values will
be lost, which will cause codec stop working. Add suspend/resume fucnion to
restore the register values.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
---
 drivers/mfd/wm8994-core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mfd/wm8994-core.c b/drivers/mfd/wm8994-core.c
index e6fab94..a33eb04 100644
--- a/drivers/mfd/wm8994-core.c
+++ b/drivers/mfd/wm8994-core.c
@@ -672,6 +672,7 @@ MODULE_DEVICE_TABLE(i2c, wm8994_i2c_id);
 
 static const struct dev_pm_ops wm8994_pm_ops = {
 	SET_RUNTIME_PM_OPS(wm8994_suspend, wm8994_resume, NULL)
+	SET_SYSTEM_SLEEP_PM_OPS(wm8994_suspend, wm8994_resume)
 };
 
 static struct i2c_driver wm8994_i2c_driver = {
-- 
1.8.0

